docs/DeveloperGuide.md

```diff
@@ -8,30 +29,194 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Main Mode
+
+This section describes the implementation of Main Mode features.
+
+Once the `main()` method of ComputerComponentChooser is called, objects for the `BuildManager`, `Parser`, 
+`editParser`, `Storage`, `Ui` are created.
+
+
+
+#### BuildManager
+
+![](/images/BuildManager.png)
+
+The builds are managed by the BuildManager class. It contains a list of builds in a HashMap. The class also contains
+methods to find and filter builds based on user requirements. The BuildManager class is a singleton class, meaning that
+there is only one instance of the class in the program. This is to ensure that there is only one list of builds in the
+program. 
+
+In our application, BuildManager is a class object that contains a HashMap of builds. The HashMap is used to store the
+builds in the program. The key of the HashMap is the name of the build, and the value is the build object.
+
+The BuildManager class is responsible for the following operations:
+- Add a build to the list of builds
+- Delete a build from the list of builds
+- Get a particular build from the list of builds
+- Get the list of builds
+- Find build that contains a search term from the list of builds
+- Filter builds based on user requirements
+
```

Comment:
slightly confused by the idea of "getting a particular build" and "finding a build that contains a search term from the list of builds" - maybe the difference could be explained more distinctly.